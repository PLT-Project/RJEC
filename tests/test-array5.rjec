struct foo {
    x int;
    y char;
    z bool;
}

func my_func (n int, c chan int) {
    var a [n]struct foo;
    for i := 0; i < n; i = i + 1 {
        printi(a[i].x);
        a[i].y = 'a';
        printb(a[i].z);
    }
    for {
        printc(a[<- c].y);
    }
}

func main() {
    c := make(chan int);
    yeet my_func(5, c);

    c <- 1;
    c <- 2;
    c <- 3;
}