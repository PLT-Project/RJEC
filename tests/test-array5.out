0
false
0
false
0
false
0
false
0
false
a
a
a